configure.ac

```diff
@@ -2387,8 +2383,16 @@ AS_IF([test "x$enable_zph_qos" = "xyes"],[
     [Enable support for QOS netfilter mark preservation])
 ])
 
-AC_CHECK_LIB(regex, regexec, [REGEXLIB="-lregex"],[REGEXLIB=''])
-AC_SUBST(REGEXLIB)
+dnl Look for libregex with regcomp() API
+SQUID_AUTO_LIB(regex,[GNU Regex],[LIBREGEX])
+SQUID_CHECK_LIB_WORKS(regex,[
+  PKG_CHECK_MODULES([LIBREGEX],[libregex],[:],[:])
+  CPPFLAGS="$LIBREGEX_CFLAGS $CPPFLAGS"
+  LIBS="$LIBREGEX_PATH $LIBREGEX_LIBS $LIBS"
+  AC_CHECK_HEADERS([regex.h],[
+    LIBREGEX_LIBS="$LIBREGEX_PATH -lregex"
```

Comment:
It looks odd that we are using LIBREGEX_LIBS to adjust LIBS and then modifying LIBREGEX_LIBS (while keeping the old value in LIBS). Why do we utilize this use-then-modify trick here?

And why does this linking-related LIBREGEX_LIBS manipulation depend on a C++ header presence (while other regex checks/adjustments do not)?